src/main/java/duke/task/Task.java

```diff
@@ -0,0 +1,136 @@
+package duke.task;
+
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+
+/**
+ * Tasks are the main data of the Duke application, describing something that
+ * needs to be done.
+ */
+public class Task {
+
+    /**
+     * Array of date formats that Duke is able to parse. A valid date can take
+     * any of the following formats:
+     * 1. EEE, dd MMM yy, hh:mm
+     * 2. dd-MM-yyyy hh:mm
+     * 3. dd-MM-yyyy
+     * 4. dd-MM-yy
+     * 5. hh:mm
+     * 6. EEE
+     * where EEE is the day of week (e.g. Mon, Fri).
+     */
+    public static final SimpleDateFormat[] DATE_FORMATS = {
+        new SimpleDateFormat("EEE, dd MMM yy, hh:mm"),
+        new SimpleDateFormat("dd-MM-yy hh:mm"),
+        new SimpleDateFormat("dd-MM-yyyy"),
+        new SimpleDateFormat("dd-MM-yy"),
+        new SimpleDateFormat("hh:mm"),
+        new SimpleDateFormat("EEE")
+    };
+
+    /** Description of task. */
+    private String description;
+
+    /** Whether the task has been completed. */
+    private boolean isDone;
+
+    /**
+     * Creates a new undone Task with the given description.
+     *
+     * @param description   Description of the Task. Description length should
+     *                      be at most 50 characters (for now).
+     */
+    public Task(String description) {
+        this(description, false);
+    }
+
+    /**
+     * Creates a new Task with the given description and isDone status.
+     *
+     * @param description   Description of the Task. Description length should
+     *                      be at most 50 characters (for now).
+     */
+    public Task(String description, boolean isDone) {
+        this.description = description;
+        this.isDone = isDone;
+    }
+
+    /**
+     * Returns a Date corresponding to the date represented by the given String.
+     * If the input format is invalid (does not match any of the date formats
+     * specified in Task.DATE_FORMATS), returns the Date representing the
+     * current instant.
+     *
+     * @param date      String representing a date
+     * @return Date     Date corresponding to the given date String, or the
+     *                  current instant
+     */
+    static Date parseDate(String date) {
+        for (SimpleDateFormat format : DATE_FORMATS) {
+            try {
+                return format.parse(date);
+            } catch (ParseException e) {
+                // do nothing and try the next format
+            }
+        }
+        return new Date();
+    }
+
+    /**
+     * Returns a plus symbol or space according to the isDone status of the
+     * current task.
+     *
+     * @return  A plus symbol if the Task is done, and a single space otherwise.
+     */
+    private String getStatusIcon() {
+        return isDone ? "+" : " ";
+        // return isDone ? "\u2713" : "\u2718"; // return tick or X symbols
```

Comment:
If I'm not wrong, outputting to a `Printstream` with UTF-8 `Charset` allows for Unicode to be printed.